02_bubble/ailyntang/sort.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+	if len(s) < 2 {
+		return s
+	}
+
+	beforeSort := make([]int, len(s))
+	copy(beforeSort, s)
+	afterSort := make([]int, len(s))
+	copy(afterSort, s)
+
+	sortAdjacentPairs(afterSort)
+
+	for i, num := range afterSort {
+		if num != beforeSort[i] {
+			copy(beforeSort, afterSort)
+			sortAdjacentPairs(afterSort)
+		}
+	}
+
+	return afterSort
+}
+
+func sortAdjacentPairs(s []int) []int {
```

Comment:
This function looks good to me. I think it does what it's meant to but it would be a good idea to write some tests for the function in isolation to be sure it does what's expected

Reply:
Done, thanks. I'll remember to add tests for all standalone functions in the future.